lib/mongo/collection.rb

```diff
@@ -350,10 +350,54 @@ def watch(pipeline = [], options = {})
     # @return [ Integer ] The document count.
     #
     # @since 2.1.0
+    #
+    # @deprecated Use count_documents or estimated_document_count instead.
     def count(filter = nil, options = {})
       View.new(self, filter || {}, options).count(options)
     end
 
+    # Get a count of matching documents in the collection.
```

Comment:
I think this should explain the difference between count_documents and the deprecated count, to prevent user confusion.